x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -145,42 +139,61 @@ export const SampleLogsInput = React.memo<SampleLogsInputProps>(({ integrationSe
         setSampleFileError(undefined);
         setIntegrationSettings({
           ...integrationSettings,
-          logsSampleParsed: undefined,
+          logSamples: undefined,
           samplesFormat: undefined,
         });
         return;
       }
 
+      if (logsSampleFile.size > MAX_IMPORT_PAYLOAD_BYTES) {
+        // File size limited to 9 MegaBytes
+        setSampleFileError(i18n.LOGS_SAMPLE_ERROR.LOGS_SAMPLE_FILE_TOO_LARGE);
+        setIntegrationSettings({ ...integrationSettings, logSamples: undefined });
+        return;
+      }
+
       const reader = new FileReader();
       reader.onload = function (e) {
         const fileContent = e.target?.result as string | undefined; // We can safely cast to string since we call `readAsText` to load the file.
-        const { error, isTruncated, logsSampleParsed, samplesFormat } =
-          parseLogsContent(fileContent);
+        if (fileContent == null) {
+          return { error: i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ };
+        }
+        let logSamples;
+        const { error, jsonSamples, samplesFormat } = parseJsonLogsContent(fileContent);
         setIsParsing(false);
-        setSampleFileError(error);
+
         if (error) {
-          setIntegrationSettings({
-            ...integrationSettings,
-            logsSampleParsed: undefined,
-            samplesFormat: undefined,
-          });
-          return;
+          if (error === i18n.LOGS_SAMPLE_ERROR.CAN_NOT_PARSE) {
+            // Non JSON Samples
+            logSamples = fileContent.split('\n').filter((line) => line.trim() !== '');
```

Comment:
This might be a nit, but any reason we would want to perform a split on the whole file, rather than slicing it first then splitting on just those few 100 lines?

Reply:
Hmm.. Since we keep this to 10 , it is not relevant anymore

Reply:
I mean its the same thing there, but we can improve it later, since we are already doing it this way 👍

Reply:
At this point , we would have `readAsText` checked if it is JSON/NDJSON and then left with a big string to handle. Hence splitting it first and then slicing to 10 rows